php/libraries/User.class.inc

```diff
@@ -66,19 +70,33 @@ class User extends UserPermissions
         $row = $DB->pselectRow($query, array('UID' => $username));
 
         // get user sites
-        $user_centerID_query =  $DB->pselect(
-            "SELECT CenterID FROM user_psc_rel upr
-                        WHERE upr.UserID= :UID",
-            array('UID' => $row['ID'])
-        );
+        $user_cid = array();
+        if (!isset($row['ID'])) {
+            error_log(
+                "Could not get user center ID(s) because'
+                . ' user DB query value 'ID' is not set."
+            );
+        } else {
+            $user_centerID_query =  $DB->pselect(
+                "SELECT CenterID FROM user_psc_rel upr
+                WHERE upr.UserID= :UID",
+                array('UID' => $row['ID'])
+            );
```

Comment:
maybe this would be a good use for @ridz1208's new pselectCol function?

Reply:
https://github.com/aces/Loris/pull/3165 is not merged yet